src/lib/core/NodeId.h

```diff
@@ -0,0 +1,60 @@
+/*
+ *
+ *    Copyright (c) 2021 Project CHIP Authors
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+#pragma once
+
+#include <cstdint>
+
+namespace chip {
+
+// TODO: Consider making this a class and the various utility methods static
+// methods.
+using NodeId = uint64_t;
+
+constexpr NodeId kUndefinedNodeId = 0ULL;
+constexpr NodeId kAnyNodeId       = 0xFFFF'FFFF'FFFF'FFFFULL;
```

Comment:
This does not exist in spec. Suggest replacing with a value in TemporaryLocalId range, (e.g., `0xFFFF'FFFE'FFFF'FFFFULL`)  and subtracting 1 from `kMaxTemporaryLocalId`

Reply:
This _does_ exist in the spec, afaict.  Unwinding the spec definitions:

* First 6 bytes are `FFFF_FFFF_FFFF`, so this is a "Group Node ID"
* Group id is `FFFF`, so this is the "All Nodes" group.

Perhaps the disagreement here is what "any node" means?

Reply:
> Perhaps the disagreement here is what "any node" means?

Yes. "Any node" is not really something I understand from the spec, and definitely the usage here is as a special value placeholder in SDK impl.

Reply:
Alright, audited some code and we can do this....